RecoMET/METAlgorithms/src/CSCHaloAlgo.cc

```diff
@@ -757,44 +639,695 @@ math::XYZPoint CSCHaloAlgo::getPosition(const DetId &id, reco::Vertex::Point vtx
+      //Selects only second HCAL layer
+      if(abs(rhposj.z())<425) continue;
+      
+      double rhetj = rechitj.energy()/cosh(rhposj.eta());
+      if(rhetj<2) continue;
+      
+      double phij = rhposj.phi();
+      if(abs(deltaPhi(phi,phij))>0.4 ) continue;
+      
+      double rhrj = sqrt(rhposj.x()*rhposj.x()+rhposj.y()*rhposj.y());
+      if(abs(rhr-rhrj)>50) continue;
+
+      h2oe+=rhetj/etcluster;
+    }
+    //H/E condition
+    if(h2oe>0.1) continue; 
+        
+
+    clustercand.SetClusterEt(etcluster); 
+    clustercand.SetSeedEt(rhet); 
+    clustercand.SetSeedEta(eta); 
+    clustercand.SetSeedPhi(phi); 
+    clustercand.SetSeedZ(rhpos.Z());
+    clustercand.SetSeedR(sqrt(rhpos.x()*rhpos.x()+rhpos.y()*rhpos.y())); 
+    clustercand.SetSeedTime(rechit.time()); 
+    clustercand.SetH2overE(h2oe);
+    clustercand.SetNbEarlyCrystals(nbcrystalssmallt);
+    clustercand.SetNbLateCrystals(nbcrystalshight);
+    clustercand.SetClusterSize(clustersize);
+    clustercand.SetTimeDiscriminator(timediscriminator);
+    clustercand.SetBeamHaloRecHitsCandidates(bhrhcandidates);
+    TheHaloClusterCandsEE.push_back(clustercand);
   }
+
+  return  TheHaloClusterCandsEE;
+} 
+
+
+
+
+ 
+
+std::vector<HaloClusterCandidateHB> CSCHaloAlgo::GetHaloClusterCandidateHB(edm::Handle<EcalRecHitCollection>& ecalrechitcoll, edm::Handle<HBHERecHitCollection>& hbherechitcoll,float et_thresh_seedrh){
+
+  std::vector<HaloClusterCandidateHB> TheHaloClusterCandsHB;
+
+  reco::Vertex::Point vtx(0,0,0);
+
+  for(size_t ihit = 0; ihit<hbherechitcoll->size(); ++ ihit){
+    HaloClusterCandidateHB  clustercand;
+    
+    const HBHERecHit & rechit = (*hbherechitcoll)[ ihit ];
+    math::XYZPoint rhpos = getPosition(rechit.id(),vtx);
+    //Et condition
+    double rhet = rechit.energy()/cosh(rhpos.eta());
+    if(rhet<et_thresh_seedrh) continue;
+    if(abs(rhpos.z())>380) continue;
+    double eta = rhpos.eta();
+    double phi = rhpos.phi();
+    
+    bool isiso = true;
+    double etcluster(0);
+    int nbtowerssameeta(0);
+    double timediscriminatorITBH(0),timediscriminatorOTBH(0);
+    double etstrip_phiseedplus1(0), etstrip_phiseedminus1(0);
+
+    //Building the cluster 
+    edm::RefVector<HBHERecHitCollection> bhrhcandidates;
+    for(size_t jhit = 0; jhit<hbherechitcoll->size(); ++ jhit){
+      const HBHERecHit & rechitj = (*hbherechitcoll)[ jhit ];
+      HBHERecHitRef rhRef(hbherechitcoll,jhit);
+      math::XYZPoint rhposj = getPosition(rechitj.id(),vtx);
+      double rhetj = rechitj.energy()/cosh(rhposj.eta());
+      if(rhetj<2) continue;
+      if(abs(rhposj.z())>380) continue;
+      double etaj = rhposj.eta();
+      double phij = rhposj.phi();
+      if(abs(eta-etaj)>0.4) continue;//This means +/-4 towers in eta 
+      if(abs(deltaPhi(phi,phij))>0.2) continue;//This means +/-2 towers in phi 
+      if(abs(deltaPhi(phi,phij))>0.1&&abs(eta-etaj)<0.2){isiso=false;break;}//The strip should be isolated
```

Comment:
please don't copy-paste function calls. (The compiler may or may not optimize these away. )
Define a temporary variable instead.